signer/src/emily_client.rs

```diff
@@ -57,30 +91,73 @@ impl TryFrom<&Url> for EmilyClient {
         }
 
         let mut config = EmilyApiConfig::new();
-        config.base_path = url.to_string();
+        // Url::parse defaults `path` to `/` even if the parsed url was without the trailing `/`
+        // causing the api calls to have two leading slashes in the path (getting a 404)
+        config.base_path = url.to_string().trim_end_matches("/").to_string();
 
         Ok(Self { config })
     }
 }
 
 impl EmilyInteract for EmilyClient {
     async fn get_deposits(&self) -> Result<Vec<CreateDepositRequest>, Error> {
-        // just to kill the unused warning for now (Self.config)
-        let _ = &self.config;
-
-        // TODO: We need to be able to build `CreateDepositRequests` from the `DepositInfo` response.
-        // However, we don't have all of the information we need yet from Emily to do that.
-        // For now, we'll just return an empty vector.
-        Ok(vec![])
-
-        // let _resp = deposit_api::get_deposits(
-        //         &self.config,
-        //         Status::Pending,
-        //         None,
-        //         None
-        //     )
-        //     .await
-        //     .map_err(EmilyClientError::GetDeposits)?;
+        // TODO: hanlde pagination -- if the queried data is over 1MB DynamoDB will
+        // paginate the results even if we pass `None` as page limit.
+        let resp = deposit_api::get_deposits(&self.config, Status::Pending, None, None)
+            .await
+            .map_err(EmilyClientError::GetDeposits)
+            .map_err(Error::EmilyApi)?;
+
+        resp.deposits
+            .iter()
+            .map(|deposit| {
+                Ok(CreateDepositRequest {
+                    outpoint: OutPoint {
+                        txid: Txid::from_str(&deposit.bitcoin_txid)
+                            .map_err(|_| Error::TypeConversion)?,
+                        vout: deposit.bitcoin_tx_output_index,
+                    },
+                    reclaim_script: ScriptBuf::from_hex(&deposit.reclaim_script)
+                        .map_err(|_| Error::TypeConversion)?,
+                    deposit_script: ScriptBuf::from_hex(&deposit.deposit_script)
+                        .map_err(|_| Error::TypeConversion)?,
```

Comment:
We probably want more specific error variants for these than the ones here.

Reply:
Done in [7200f63](https://github.com/stacks-network/sbtc/pull/609/commits/7200f633c564223600c5867dfb13a721d278bc77)